support-frontend/assets/helpers/paymentIntegrations/readerRevenueApis.js

```diff
@@ -40,27 +40,37 @@ export type StripePaymentMethod = 'StripeCheckout' | 'StripeApplePay' | 'StripeP
 export type StripePaymentRequestButtonMethod = 'none' | StripePaymentMethod;
 
 type RegularContribution = {|
+  productType: 'Contribution',
   amount: number,
   currency: string,
   billingPeriod: BillingPeriod,
 |};
 
 export type DigitalSubscription = {|
+  productType: 'DigitalSubscription',
   currency: string,
   billingPeriod: BillingPeriod,
   readerType: ReaderType,
-  giftMessage?: string | null,
-  giftDeliveryDate?: string | null,
 |};
 
 export type PaperSubscription = {|
+  productType: 'Paper',
```

Comment:
I would suggest using the types from `SubscriptionProduct` here
```suggestion
  productType: typeof Paper,
```

It will simplify code lower down